### ISSUE TITLE
[BUG]: cannot sync with proxy

### ISSUE THREAD

**Issue by ascpial** [ankidroid/Anki-Android#14478]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Use the app on a network that requires a proxy, for example a school network.

### Expected behaviour

The app sync as usual without issues.

### Actual behaviour

On such networks, the application is unable to sync (the sync popup stays still), and stopping the synchronization does freeze the app.

https://github.com/ankidroid/Anki-Android/assets/72463129/64140fe5-4b66-464d-b46a-45dfe495fb13

### Debug info

```text
AnkiDroid Version = 2.16.5 (953b9bd879269910ee962b520da9705336d8bc2d)

Android Version = 13

ProductFlavor = full

Manufacturer = Google

Model = Pixel 6a

Hardware = bluejay

Webview User Agent = Mozilla/5.0 (Linux; Android 13; Pixel 6a Build/TQ3A.230901.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/117.0.0.0 Mobile Safari/537.36

ACRA UUID = 053762f1-9544-42c4-b1f6-66c7044b738e

New schema = true

Scheduler = std3

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

I know that #3787 is related, but this Github issue is old and I didn't wanted to reopen it.

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
I gave a good effort at reproducing this on the latest alpha and I was unable to (for anyone curious, I was using Proxyman.io app to create a local proxy then sending network traffic through it)

Can you see if this still reproduces in the latest 2.17 alphas? https://github.com/ankidroid/Anki-Android/releases

If it does, I'll need to know more about your network layout - is the proxy a transparent thing on the network gateway and your traffic is forced through there? Do you manually configure it? Does it handle HTTP and HTTPS?

Going to bump this to the 2.18 milestone until we know more about it + can reproduce

**Comment by david-allison** [comment]
For when we get around to this: We likely want a developer option to block the sync. It's difficult to test currently

**Comment by mikehardy** [comment]
@david-allison do you have an apple developer account? I have Network Link Conditioner from the "Additional Tools" download related to Xcode (https://developer.apple.com/download/all/?q=Additional%20Tools) and with that I can make my macOS network reaaaaallllllyyy slow (and/or lossy) which gives me time to test this.

I'd rather use something like this vs adding some complications

**Comment by david-allison** [comment]
I'll limit my network for now

For later: I prefer testing syncing on my phone as it retains account data from the password manager

Very minor, but testing syncing isn't the best developer experience and I'll take every win I can get

**Comment by mikehardy** [comment]
ok - your test results match mine @ascpial thanks for reporting back - I'm going to close this as apparently fixed
If this recurs for some reason we can reopen

**Comment by ascpial** [comment]
I'm sorry, I made a spelling mistake.
There sould have been a "not" in the second sentence.
The app is usable with the proxy on but the sync does not works.

Sorry for this mistake...

**Comment by izogfif** [comment]
Here is a workaround for those who have to use proxy (or custom HTTPS / SSL / TLS certificate). Let's say you need to sync with Anki sync server http://anki.local and it's only available via HTTP proxy running on address 192.168.1.1 on port 3128:
* Setup HTTP-server, let's say, nginx, on some host that's visible from the device running AnkiDroid. For example, on http://192.168.1.2:4600.
* Configure this server to accept all incoming requests on port 4600 and pass them to the target Anki sync server (http://anki.local). You can configure nginx use HTTP proxy running on 192.168.1.1 port 3128 as specified in [this StackOverflow answer](https://stackoverflow.com/a/46824465/156973).
* Specify http://192.168.1.2:4600 as your Anki sync server URL in AnkiDroid settings.

It'll work like this: AnkiDroid connects to http://192.168.1.2:4600. nginx forwards all requests to http://anki.local using proxy_pass configuration:

```
worker_processes auto;

events {
  worker_connections 1024;
  multi_accept on;
}

http {
  include mime.types;

  server {
    location / {
      proxy_pass http://anki.local;
      proxy_set_header Host anki.local;
    }
  }
} 
```
 
Note: if you need proxy only because your device running AnkiDroid can't resolve DNS of http://anki.local, but the host you're running nginx on can, then you might not need to use that StackOverflow answer at all!
